### ISSUE TITLE
Issue with sosie3.x interpolation: undefined symbol: __netcdf_MOD_nf90_open

### ISSUE THREAD

**Issue by SOFIADAL** [brodeau/sosie#16]
Hello.

I am new to programming and I just tried to interpolate an ERA5 file towards a CMEMS grid file using sosie3.x interpolation tool. (with grid types are regular). However, I get an error message like this:
[Namelist_Final.pdf](https://github.com/brodeau/sosie/files/3654728/Namelist_Final.pdf)

./sosie3.x: symbol lookup error: ./sosie3.x: undefined symbol: __netcdf_MOD_nf90_open

Namelistm seems to work fine but as soon as the interpolation is about to start this error message appears. I attach also the namelist file that I use and I copy and paste the output when I try to run sosie3.x:

==========================================================
             S  O  S  I  E    version 3.0
 ==========================================================
 
  * namelist = namelist

 Opening namelist "namelist"
 
 The current field is NOT a component of a vector!
 
 Source grid is declared as regular
 
 Target grid is declared as regular
 
 Source file: /home/sofiad/SWR_MARCH_2D.nc
 
 Method for 2D interoplation: bilin
 
 Time record name in source file: time
 
 File containing the source grid: /home/sofiad/SWR_MARCH_2D.nc
 
 Longitude name: longitude
 
 Latitude name: latitude
 
 Source grid mask: missing_value
 
 Source mask variable: 
 
 Variable to be treated:msnswrf
 
 Level to treat:           1
 
 Name of variable on the output file:temp
 
 Units of variable to be treated:
 
 Long name of variable to be treated:
 
 Target grid: /home/sofiad/CMEMS_thetao_depth_0_2017-03.nc
 
 Longitude on target grid: longitude
 
 Latitude on target grid: latitude
 
 Longitude name in target file: longitude
 
 Latitude name in target file: latitude
 
 Target land-sea mask file: /home/sofiad/mask_CMEMS_2D_FIN.nc
 
 Name of land-sea mask variable on target grid: thetao
 
 Output file: ./temp_regular-regular_monthly_1998.nc
 
 Shall we use DROWN on source fields:  T
 
  => about DROWN action:
     -> continental penetration in # of pixels:         100
     -> # of time to apply a smoothing on continents after DROWN:          50
     -> is the mask on the source field is changing with time : F
 
 East west periodicity of source grid:            0
 
 Masking output file:  T
 
 Value for masked points in output file:   -9999.00000    
 
 East west periodicity of target grid:            2
 
 Time is not controlled and will be the same than in source file! 
 
                 ---------------------
 
 Starting interpolation for config "regular-regular"...
 
./sosie3.x: symbol lookup error: ./sosie3.x: undefined symbol: __netcdf_MOD_nf90_open

Could you please help me with this?
Thank you in advance for your reply,
Kind regards,
Sofi

**Comment by raphaeldussin** [comment]
it looks to me like sosie cannot find the netcdf library.
can you be more specific on your system and how you compiled sosie?

**Comment by SOFIADAL** [comment]
Hello. Thanks for your answer.
So here are the details of my system (I am trying to run sosie3.x on a remote Machine)

NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"

Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04

And I compiled sosie following the instruction given in the frontpage of the GitHub about sosie3.x.

Thank you,
Sofi

**Comment by raphaeldussin** [comment]
you may want to add the path to the libnetcdf to the environment variable LD_LIBRARY_PATH,
you should be able to get that path with ```nf-config --flibs```

then just add to your .bashrc

```export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/path/to/libnetcdf" ```

**Comment by SOFIADAL** [comment]
Ok.thank you very much. I will let you know if this worked once I am back on my computer.

Sofi

**Comment by SOFIADAL** [comment]
Your proposition has worked indeed. Thank you. However, there seems to be a problem with the type of the mask file that I use and I get a following error like this:
NetCDF: Numeric conversion not representable.

What could that mean for the mask file? Is there a specific type of mask file that I have to use?

Thank you in advance for your help,
Kind regards,
Sofi

**Comment by raphaeldussin** [comment]
Maybe start with the examples that are included and make sure everything works. Then you can compare with your files and find differences in types,...

**Comment by SOFIADAL** [comment]
With the mask file I used, for some reason I get an interpolated file, which has an actual range of a variable: -9995.f, 316.3625f.Could that be related to the mask file that I used? 

For the file containing the target land-sea mask I have chosen this: cf_lsm_trg = 'missing_value', (whereas the attribute in my file is actually called "missing_values".), while for the "cf_x_trg" I have indicated the actual mask file. 

Also I was also wondering what the "ewper_trg", east-west periodicity means? I am confused by the description in the namelist.

Thank you in advance for your time and help,
Kind regards,
Sofi
